engine/src/flutter/impeller/display_list/canvas.h

```diff
@@ -261,6 +261,12 @@ class Canvas {
   // Visible for testing.
   bool SupportsBlitToOnscreen() const;
 
+  /// @brief Retrieve the backing texture from the DL image.
+  ///
+  /// If this image is a deferred image, automatically performs the upload.
+  std::shared_ptr<Texture> GetOrUploadTexture(
```

Comment:
Seems like we should make `impeller_texture()` private too since everyone should be calling this instead.

Reply:
I was thinking about this too but I'm not sure the right way to solve it. If I make this private but friend to Canvas, then that doesn't really stop the canvas from accessing it incorrectly.

I think maybe removing the getter and replacing it with a method that takes some kind of generic loader callback might work. WIP